src/lib/php/Data/Highlight.php

```diff
@@ -202,12 +202,12 @@ public function getHtmlElement()
   public function getArray()
   {
     return array(
-      "start" => $this->start,
-      "end" => $this->end,
-      "type" => $this->type,
+      "start" => intval($this->start),
+      "end" => intval($this->end),
+      "type" => $this->type == null ? Highlight::UNDEFINED : $this->type,
       "licenseId" => $this->licenseId,
-      "refStart" => $this->refStart,
-      "refEnd" => $this->refEnd,
+      "refStart" => $this->refStart == -1 ? 0 : $this->refStart,
+      "refEnd" => $this->refEnd == -1 ? 0 : $this->refEnd,
```

Comment:
`refStart` and `refEnd` will return `-1` for one shot highlights which I am assuming is not the desired output (lmk if this is wrong, will revert the changes). Since I have enforced checking rules in the `getArray()` function, will it still break code?

Reply:
This function is called only in API controller.